.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,44 @@
+name: Build on Push and PR
+on:
+  push:
+    branches:
+      - master
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v3
+      with:
+        distribution: temurin
+        java-version: 11
+        
+    - name: Setup Gradle
+      uses: gradle/gradle-build-action@v2
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 12
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v2
+      with:
+        platforms: linux/amd64,linux/arm64
+
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: Login to DockerHub
+      if: github.event_name != 'pull_request'
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
```

Comment:
Env vars available are named:
`DOCKER_HUB_REGISTRY_USERNAME` and `DOCKER_HUB_REGISTRY_PASSWORD `.